deployments/databricks/deploy_collector.sh

```diff
@@ -0,0 +1,170 @@
+# Copyright Splunk Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+#!/bin/bash
+
+set -euo pipefail
+set -x
+
+# This script is used to deploy the Splunk distribution of the OpenTelemetry Collector
+# on the current node of a Databricks cluster. Through UI configuration the script will
+# be distributed and run on every node of the cluster.
+
+# Required Variables:
+# - SPLUNK_ACCESS_TOKEN: Splunk o11y access token for sending data to Splunk o11y backend.
+# - DATABRICKS_CLUSTER_HOSTNAME: Hostname of the Databricks compute resource. Use the Server
+#   Hostname described here:
+#   https://docs.databricks.com/en/integrations/compute-details.html
+# - DATABRICKS_ACCESS_TOKEN: Databricks personal access token (PAT) used to connect to the Apache Spark API.
+#   Directions for creating a PAT: https://docs.databricks.com/en/dev-tools/auth/pat.html
+
+# Optional Variables:
+# - OTEL_VERSION: Version of the Splunk OpenTelemetry Collector to deploy as a part of this release.
+#   Default: "latest". Valid version must be >=0.120.0.
+# - SCRIPT_DIR: Installation path for the Collector and its config
+# - SPLUNK_REALM: Splunk o11y realm to send data to. Default: us0
+
+DATABRICKS_CLUSTER_HOSTNAME=${DATABRICKS_CLUSTER_HOSTNAME:-}
+SPLUNK_ACCESS_TOKEN=${SPLUNK_ACCESS_TOKEN:-}
+DATABRICKS_ACCESS_TOKEN=${DATABRICKS_ACCESS_TOKEN:-}
+
+if [ -z "${DATABRICKS_CLUSTER_HOSTNAME}" ]; then
+  echo "environment variable 'DATABRICKS_CLUSTER_HOSTNAME' must be set, exiting."
+  exit 1
+fi
+
+if [ -z "${SPLUNK_ACCESS_TOKEN}" ]; then
+  echo "environment variable 'SPLUNK_ACCESS_TOKEN' must be set, exiting."
+  exit 1
+fi
+
+if [ -z "${SDATABRICKS_ACCESS_TOKEN}" ]; then
+  echo "environment variable 'DATABRICKS_ACCESS_TOKEN' must be set, exiting."
+fi
+
+
+OTEL_VERSION=${OTEL_VERSION:-latest}
```

Comment:
Happy to discuss variable names here, I'm sure they could be made more clear and obvious.

Reply:
I see a few different variations in use `SPLUNK_OTEL_VERSION` (TA), `SPLUNK_OTEL_COLLECTOR_VERSION` (option in puppet role) and also `OTEL_VERSION` (cloudfoundry). I think prefixing with SPLUNK makes sense cause we might push patch versions which does not match with a version on upstream. Wdyt?

Reply:
Good call. I've updated all variables that are Splunk-specific (release version, URLs, binary file names, etc.) to include the `SPLUNK_` prefix.